cdap-common/bin/functions.sh

```diff
@@ -493,11 +491,34 @@ cdap_set_spark() {
         return 1
       fi
       SPARK_HOME=$(echo -e "${SPARK_VAR_OUT}" | grep ^SPARK_HOME= | cut -d= -f2)
-      export SPARK_HOME
-      return 0
     fi
-    return 1
   fi
+
+  export SPARK_HOME
+
+  # Find environment variables setup via spark-env.sh
+  SPARK_ENV_SH=${SPARK_HOME}/conf/spark-env.sh
+  if [[ -f ${SPARK_ENV_SH} ]]; then
+    if [[ -z ${SPARK_ENV_PATTERN} ]]; then
+      # By default, only get env from spark-env.sh that starts with PY or SPARK
+      SPARK_ENV_PATTERN="^(PY|SPARK)"
+    fi
+    SPARK_ENV=$(source ${SPARK_ENV_SH}; env | grep -E "${SPARK_ENV_PATTERN}")
```

Comment:
This can (basically) be replaced with:
```
SPARK_ENV=$(echo -e "${SPARK_VAR_OUT}" | grep -E "${SPARK_ENV_PATTERN}")
```

It's no longer acceptable to just set `SPARK_HOME` even for MapR, so the assignment of `SPARK_VAR_OUT` needs to be done, unconditionally. In that case, you already have the entire Spark execution environment stored in `SPARK_VAR_OUT` and it's fully resolved by the Spark-shipped scripts, correctly.